tests/ga/test_update.py

```diff
@@ -1740,6 +1741,66 @@ def test_it_should_recreate_extension_event_directories_for_existing_extensions_
                 for ext_dir in expected_events_dirs:
                     self.assertTrue(os.path.exists(ext_dir), "Extension directory {0} should exist!".format(ext_dir))
 
+    def test_it_should_report_update_status_in_status_blob(self):
+        with _get_update_handler(iterations=1) as (update_handler, protocol):
+            with patch.object(conf, "get_autoupdate_gafamily", return_value="Prod"):
+                with patch("azurelinuxagent.common.logger.warn") as patch_warn:
+
+                    protocol.aggregate_status = None
+                    protocol.incarnation = 1
+
+                    def mock_http_put(url, *args, **_):
```

Comment:
this feels like it could break if different types of PUT calls are added to the agent; there are only two choices below and one of them is assumed (i.e. not explicitly checked for, it's just in  the equivalent of an `else` clause.

I think this could be more or less fixed by making the second scenario explicit (via an if statement) and then do a `return None` to ignore anything else.

Reply:
That's a good point, we should indeed make it explicit. That being said, unfortunately, as per the current code, its very hard to differentiate if a PUT call is for status or not because we directly upload to the storage blob uri we get from GS - 

> if self.status_blob.upload(extensions_goal_state.status_upload_blob):

The only other differentiating factor we have is a PUT call. One way to deal with it would be to change all of our test files and add a unique string in the `<StatusUploadBlob>` object to specify that its a status blob and then filter it out based on those, but for the time being, since we dont have any other PUT calls, I'd rather keep it as is to avoid the extra trouble :)

Reply:
Could you just create a new GS file to use for this test and edit it such that the storage blob uri matches some regex (or just straight up is some string) and filter off that?

That would still be a bit fragile (i.e. someone changing the specific file you used might break things) but I feel like it's a good compromise between leaving it as is and doing the extra work of manipulating the status blob string itself.

Reply:
After thinking about this again, since its being used in multiple places, I'd rather fix this in a separate PR in all those usages.